packages/bpk-component-carousel/src/BpkCarousel.js

```diff
@@ -0,0 +1,195 @@
+  prevArrowDisabled: boolean,
+  nextArrowDisabled: boolean,
+};
+
+class BpkCarousel extends Component<Props, State> {
+  wrapperRef: Object;
+
+  firstItemRef: Object;
+
+  wrapperWidth: number;
+
+  itemWidth: number;
+
+  static defaultProps = {};
+
+  constructor(props: Props) {
+    super(props);
+
+    this.wrapperRef = createRef();
+    this.firstItemRef = createRef();
+
+    this.state = {
+      position: 0,
+      prevArrowDisabled: true,
+      nextArrowDisabled: false,
+    };
+  }
+
+  // Here we are getting the widths of elements to use when doing translations
+  componentDidMount() {
+    const { position } = this.state;
+    const { children } = this.props;
+    this.wrapperWidth =
+      this.wrapperRef && this.wrapperRef.current
+        ? this.wrapperRef.current.offsetWidth
+        : 0;
+    this.itemWidth =
+      this.firstItemRef && this.firstItemRef.current
+        ? this.firstItemRef.current.offsetWidth
+        : 0;
+    const itemsShown = Math.floor(this.wrapperWidth / this.itemWidth);
+    this.setState({
+      nextArrowDisabled:
+        position === Children.count(children) - itemsShown ||
+        Children.count(children) === itemsShown,
+      prevArrowDisabled: position === 0,
+    });
+  }
+
+  // Function to do the animation based on the item size that is moving
+  getTranslate = () => {
+    const { position } = this.state;
+    return `calc((-${this.itemWidth}px) * ${position})`;
+  };
+
+  // Function that will handle moving to the next slide
+  nextSlide = () => {
+    const { position } = this.state;
+    const { children } = this.props;
+    const numItems = Children.count(children) || 1;
+    this.doSliding(position === numItems - 1 ? 0 : position + 1);
+  };
+
+  // Function that will handle moving to the previous slide
+  prevSlide = () => {
+    const { position } = this.state;
+    const { children } = this.props;
+    const numItems = Children.count(children) || 1;
+    this.doSliding(position === 0 ? numItems - 1 : position - 1);
+  };
+
+  // Handles the settings of the slider position and calculating if the bottons should be disabled
+  doSliding = (position: number) => {
+    const { children } = this.props;
+    const itemsShown = Math.floor(this.wrapperWidth / this.itemWidth);
+    this.setState({
+      position,
+      nextArrowDisabled:
+        position === Children.count(children) - itemsShown ||
+        Children.count(children) === itemsShown,
+      prevArrowDisabled: position === 0,
+    });
+  };
+
+  render() {
+    const {
+      className,
+      wrapperClassName,
+      itemClassName,
+      children,
+      ...rest
+    } = this.props;
+    const classNames = getClassName('bpk-carousel', className);
+    const wrapperClassNames = getClassName(
+      'bpk-carousel__wrapper',
+      wrapperClassName,
+    );
+    const itemClassNames = getClassName('bpk-carousel__item', itemClassName);
+
+    return (
+      <div className={classNames} {...rest}>
+        <div>
```

Comment:
Do we need this extra div?

Reply:
Yes otherwise the buttons stretch the full height of the carousel so this keeps it at the correct size